### PR TITLE
fix: fixed counter exceding length of text

### PR DESCRIPTION
**Pull request by vidriduch**
- Fixed counter for skipping text exceeding total length of text
  when key was pressed on last line

### CODE REVIEW

File: levels/utils.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func PrintText(text string, pretty_print bool) {
 	var counter = 0
 	for _, line := range lines {
 		counter += len(line) + 1
+		if counter > len(text) {
```
Comment (mrshu) [inline levels/utils.go:85]: @Foxiy is it crazy to have tests for this?

**Comment by mrshu** [comment]
Given that this bug was quite a big one I believe we need a way of avoiding it in the future. I do not see other ways of doing that than having a test suite, that would test this functionality (and also this fix).

I believe that this change (small in size by huge in terms of pain it could save us from) might be a good starting point. In all honesty, I do not even know how to asses (in other words test) whether this change even fixes the issue (since you marked it as a fix).

**Comment by vidriduch** [comment]
@mrshu added new file that could server as test for all output and print line function testing and added test for this bug.

File: levels/utils.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ func print_line(text string, keypress chan []byte, echo_state bool) {
 	var counter = 0
 	var b []byte = make([]byte, 1)
 	for _, char := range text {
-		print(string(char))
+		fmt.Printf("%s", string(char))
```
Comment (mrshu) [inline levels/utils.go:28]: @Foxiy I assume this was necessary but seems like a win anyway, since we were already using `fmt` throughout the file.

**Comment by mrshu** [comment]
@Foxiy Awesome, thanks a ton for taking the floor and putting together a test for this fix, it makes me feel way better when I click on that big green button.

Thanks!

**Pull request merged**